### PR TITLE
improve the qmk setup flow

### PR DESCRIPTION
**Pull request by skullydazed**
Improve how `qmk setup` works by allowing people to specify a destination without specifying a fork. It also offers to set the location for `qmk_firmware` in `user.qmk_home`.

This should make it easier for people who want to put `qmk_firmware` somewhere other than their home directory.

### CODE REVIEW

File: qmk_cli/git.py [hunk 1]
```diff
@@ -28,5 +28,5 @@ def clone(url, destination, branch):
         cli.log.error('git clone exited %d', p.returncode)
         return False
```
Comment (Erovia) [inline qmk_cli/git.py:29]:
```suggestion

```

File: qmk_cli/git.py [hunk 2]
```diff
@@ -28,5 +28,5 @@ def clone(url, destination, branch):
...
     else:
-        cli.log.info('Successfully cloned %s to %s!', cli.args.fork, cli.args.destination)
+        cli.log.info('Successfully cloned %s to %s!', url, destination)
         return True
```
Comment (Erovia) [inline qmk_cli/git.py:32]:
```suggestion
    return p.returncode
```
I'd be in favour of forwarding git's rc on errors with `exit(rc)`.

File: qmk_cli/subcommands/setup.py [hunk 3]
```diff
@@ -18,42 +19,40 @@
...
+    if (cli.args.home / 'Makefile').exists():
+        cli.log.info('Found qmk_firmware at %s.', str(cli.args.home))
     else:
-        cli.log.error('qmk_firmware not found!')
-        if question('Would you like to clone %s?' % cli.args.fork):
+        cli.log.error('Could not find qmk_firmware!')
+        if question(clone_prompt):
             git_url = '/'.join((cli.config.setup.baseurl, cli.args.fork))
-            clone(git_url, cli.args.destination, cli.config.setup.branch)
+            clone(git_url, cli.args.home, cli.config.setup.branch)
```
Comment (Erovia) [inline qmk_cli/subcommands/setup.py:43]:
```suggestion
            rc = clone(git_url, cli.args.home, cli.config.setup.branch)
            if rc != 0:
                exit(rc)
```
Do not write the config file if cannot clone the repo.

Sadly, github won't let me comment on untouched files, but I'd change `clone.py` too:
```
-    success = git_clone(git_url, cli.args.destination, cli.config.clone.branch)
-    exit(0 if success else 1)
+    rc = git_clone(git_url, cli.args.destination, cli.config.clone.branch)
+    exit(rc)
```

File: qmk_cli/subcommands/setup.py [hunk 4]
```diff
@@ -18,42 +19,40 @@
...
+
+    # Offer to set `user.qmk_home` for them.
+    if cli.args.home != os.environ['QMK_HOME'] and question(home_prompt):
+        cli.config['user']['qmk_home'] = str(cli.args.home)
```
Comment (Erovia) [inline qmk_cli/subcommands/setup.py:47]:
```suggestion
        cli.config['user']['qmk_home'] = str(cli.args.home.absolute())
```
Relative paths will not work reliably and that's what the users will most likely provide.

File: qmk_cli/subcommands/setup.py [hunk 5]
```diff
@@ -18,42 +19,42 @@
...
+        cli.log.info('Found qmk_firmware at %s.', str(cli.args.home))
     else:
-        cli.log.error('qmk_firmware not found!')
-        if question('Would you like to clone %s?' % cli.args.fork):
+        cli.log.error('Could not find qmk_firmware!')
+        if question(clone_prompt):
             git_url = '/'.join((cli.config.setup.baseurl, cli.args.fork))
-            clone(git_url, cli.args.destination, cli.config.setup.branch)
+            result = clone(git_url, cli.args.home, cli.config.setup.branch)
+            if not result:
```
Comment (skullydazed) [inline qmk_cli/subcommands/setup.py:44]:
```suggestion
            if result != 0:
```

*innocent whistling*

File: qmk_cli/subcommands/setup.py (hunk 3 above)
Comment (Erovia) [reply qmk_cli/subcommands/setup.py]: ^ I'd update `clone.py` as well.

File: qmk_cli/subcommands/setup.py [hunk 6]
```diff
@@ -18,42 +19,44 @@
...
-            cli.log.error('Your build environment is not setup completely.')
-
-            if question('Would you like to run util/qmk_install?'):
-                curdir = os.getcwd()
-                os.chdir(str(qmk_firmware))
-                process = subprocess.run(['util/qmk_install.sh'])
-                os.chdir(curdir)
-                if process.returncode == 0:
-                    cli.log.info('QMK setup complete!')
+    qmk_bin = cli.args.home / 'bin/qmk'
```
Comment (Erovia) [inline qmk_cli/subcommands/setup.py:53]: I guess we could
```suggestion
    qmk_bin = cli.args.home / 'bin' / 'qmk'
```
, but it changes nothing.

**Pull request merged**